### PR TITLE
feat: add lint scripts workflow

### LINKED ISSUE #257: add linter for scripts
add golangci-lint to run linters on `scripts`

### PR DESCRIPTION
**Pull request by apeabody**
**What this PR does / why we need it**:
Adds `golangci-lint` workflow for scripts and required code changes for initial pass

**Which issue(s) does this PR fix**:
Fixes #257

### CODE REVIEW

File: .github/workflows/scripts.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      matrix:
+        folder: [artifacthub, require-sync, validate, website]
+    steps:
+      - uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+        with:
+          go-version: '1.19'
+          cache: false
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@08e2f20817b15149a52b5b3ebe7de50aff2ba8c5 # v3.4.0
```
Comment (sozercan) [inline .github/workflows/scripts.yaml:30]: do we want to copy .golangci-lint from GK so we have the same linters?

Comment (apeabody) [reply .github/workflows/scripts.yaml:31]: Sure thing - Looks like the existing code will need some additional work for the larger set of linters.

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  gocritic:
+    enabled-tags:
+    - performance
+  gosec:
+    excludes:
+    - G108
+  importas:
+    no-unaliased: true
+    alias:
+      - pkg: "github.com/open-policy-agent/frameworks/constraint/pkg/client"
```
Comment (sozercan) [inline .golangci.yaml:16]: do we need this for library?

File: .golangci.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+run:
+  timeout: 5m
+  skip-files:
+    - pkg/target/matchcrd_constant.go
```
Comment (sozercan) [inline .golangci.yaml:4]: same here

File: .github/workflows/scripts.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  pull_request:
+    branches:
+      - master
+    paths:
+      - ".github/workflows/scripts.yaml"
+      - "scripts/**"
+permissions:
+  contents: read
+
+concurrency:
```
Comment (sozercan) [inline .github/workflows/scripts.yaml:12]: this is good to know 👍

File: .golangci.yaml (hunk 3 above)
Comment (apeabody) [reply .golangci.yaml]: Sure, this was carry over, we can remove.

File: scripts/artifacthub/hub.go [hunk 5]
```diff
@@ -83,17 +83,17 @@ type DefaultClient struct{}
 
 // Get is a method of DefaultClient that makes the actual HTTP GET request.
 func (c DefaultClient) Get(url string) (*http.Response, error) {
-	return http.Get(url)
+	return http.Get(url) //nolint
```
[the remark this answers is not in the document]
Comment (apeabody) [reply scripts/artifacthub/hub.go:86]: Hi @nilekhc  - It is raising that the url is a variable (constructed using a const and the template), rather that their recommended best practice of only using a const for urls.

File: .golangci.yaml (hunk 2 above)
Comment (apeabody) [reply .golangci.yaml:14]: Hi @sozercan - Do we need anything else on this?  Thanks!

Comment (sozercan) [reply .golangci.yaml:14]: oh i didn't realize library used this. resolving

File: .github/workflows/scripts.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+        with:
+          go-version: '1.20'
+          cache: false
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@08e2f20817b15149a52b5b3ebe7de50aff2ba8c5 # v3.4.0
+        with:
+          version: v1.51.2
```
Comment (sozercan) [inline .github/workflows/scripts.yaml:33]:
```suggestion
          version: v1.54.2
```

**Review by sozercan — APPROVED** [review]

**Pull request merged**